pkg/types/compactor.go

```diff
@@ -56,6 +58,7 @@ func (c *CompactorConfig) AddFlags(fs *flag.FlagSet) {
 	fs.BoolVar(&c.NeedDefragmentation, "defragment", c.NeedDefragmentation, "defragment after compaction")
 	fs.DurationVar(&c.SnapshotTimeout.Duration, "etcd-snapshot-timeout", c.SnapshotTimeout.Duration, "timeout duration for taking compacted full snapshots")
 	fs.DurationVar(&c.DefragTimeout.Duration, "etcd-defrag-timeout", c.DefragTimeout.Duration, "timeout duration for etcd defrag call during compaction.")
+	fs.StringVar(&c.FullSnapshotLeaseName, "full-snapshot-lease-name", c.FullSnapshotLeaseName, "full snapshot lease name")
```

Comment:
Why do we need that option for the compactor?

Reply:
The compaction job would just call the `compact` subcommand of `etcdbrctl`. If druid deploys a full-snapshot-lease with a unique name , then druid would need a way to pass on the lease name so that the correct one is updated after compaction
[here](https://github.com/gardener/etcd-backup-restore/pull/382#discussion_r718566805) and [here](https://github.com/gardener/etcd-backup-restore/pull/382#discussion_r719575390)

Reply:
Understood, thanks for explaining.
I wonder if we need to update the `delta-snapshot-lease` as well after the compaction. At the moment it only happens when the next snapshot is taken, right? This can be too late because Druid might already start another compaction job based on an "old" delta revision number. (/cc @shreyas-s-rao)

Reply:
Yes, that is right. `delta-snapshot-lease` would only be updated when the next delta snapshot is taken
My understanding here is that druid will only react to the `delta-snapshot-lease` in the event of an update in the spec. So, druid would not start a new compaction job with the old delta revision number.

Reply:
Hm, that won't work if Druid is restarted for some reason (e.g. it is scaled by VPA). Then Druid will need to consider the `create` event for the Lease and doesn't know if it already started a compaction job or not.

Reply:
After our out-of-band discussions, added a functionality to update delta snapshot on compaction as well